### ISSUE TITLE
Handle the case when rule test is a function

### ISSUE THREAD

**Issue by sleewoo** [kisenka/svg-sprite-loader#169]
**Do you want to request a feature, report a bug or ask a question?**

report a bug

**What is the current behavior?**

it assumes all rules are strings or regexes

**What is the expected behavior?**

check whether a rule test is a function and call it rather than calling `ruleMatcher`

**If the current behavior is a bug, please provide the steps to reproduce, at least part of webpack config with loader configuration and piece of your code.**

here is a piece of webpack config that makes svg-sprite-loader to fail

```js
  {
    test(path) {
      return !path.match(/\.ts$/) || path.match(components_regex)
        ? false
        : true
    },
    exclude: /node_modules/,
    loader: 'awesome-typescript-loader',
    query: { transpileOnly: true }
  }
```

**Please tell us about your environment:**

- Node.js version: any
- webpack version: 3
- svg-sprite-loader version: 3
- OS type & version: any

**Other information (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)**

a simple fix:

in `lib/utils/get-matched-rules.js` file 

replace

```js
const matchedRules = rules.filter(rule => ruleMatcher(rule, request));
```

with

```js
  const matchedRules = rules.filter(rule =>
    typeof rule.test === 'function'
      ? rule.test(request)
      : ruleMatcher(rule, request)
  );
```

**Comment by kisenka** [comment]
@sleewoo thanks and sorry for ignoring your PR. Published as `svg-sprite-loader@3.1.1`. Ping me strongly next time when you'll need quick assistance.

**Comment by sleewoo** [comment]
thanks
